components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/ApisApiServiceImpl.java

```diff
@@ -143,4 +146,36 @@ public Response unDeployedAPIRevision(UnDeployedAPIRevisionDTO unDeployedAPIRevi
                 unDeployedAPIRevisionDTO.getEnvironment());
         return Response.ok().build();
     }
+
+    @Override
+    public Response apisApiIdOperationPolicyIsAttachedGet(String xWSO2Tenant, String apiId, String organizationId,
+                                                          MessageContext messageContext) throws APIManagementException {
+        try {
+            String organization;
+            if (organizationId != null) {
+                organization = organizationId;
+            } else {
+                organization = SubscriptionValidationDataUtil.validateTenantDomain(xWSO2Tenant, messageContext);
+            }
+            APIProvider apiProvider = RestApiCommonUtil.getLoggedInUserProvider();
+            org.wso2.carbon.apimgt.api.model.API api = apiProvider.getAPIbyUUID(apiId, organization);
+            boolean isAttached = false;
+            Set<URITemplate> uriTemplates = api.getUriTemplates();
+            for (URITemplate uriTemplate : uriTemplates) {
+                if (uriTemplate.getOperationPolicies().size() > 0) {
+                    isAttached = true;
+                    break;
+                }
+            }
+            OperationPolicyAttachmentStatusDTO result = new OperationPolicyAttachmentStatusDTO();
+            result.isAttached(isAttached);
+            return Response.ok().entity(result).build();
+
+        } catch (APIManagementException e) {
+            RestApiUtil.handleBadRequest(
+                    "An error has occurred while getting API for the given organization/tenantDomain and apiId combination.",
+                    e, log);
+        }
+        return Response.ok().build();
```

Comment:
If policies are not attached, let's return `{isAttached:false}`

Reply:
currently, it does return `{isAttached:false}`